### PR TITLE
Revendor installer, repair OpenStack MachinePool UT

### PR DESCRIPTION
**Pull request by 2uasimojo**
* Revendor installer to pick up https://github.com/openshift/installer/pull/8187 and https://github.com/openshift/installer/pull/8209
* Refactor OpenStack MachinePool code to use the new code path for trunk support discovery and accommodate the new signature of the `MachineSets()` generator
* Repair long-broken OpenStack MachinePool unit tests, which were previously blocked by the above
* Accommodate https://github.com/openshift/installer/pull/8006 which is also brought in by the revendor

A subsequent PR will use this to
* Expand unit tests to reproduce the error in the referenced card
* Fix it.

Part of [HIVE-2476](https://issues.redhat.com//browse/HIVE-2476)

### CODE REVIEW

File: pkg/controller/machinepool/openstackactuator.go [hunk 1]
```diff
@@ -45,9 +46,10 @@ func NewOpenStackActuator(masterMachine *machinev1beta1.Machine, kubeClient clie
 		return nil, err
 	}
 	actuator := &OpenStackActuator{
-		logger:     logger,
-		osImage:    osImage,
-		kubeClient: kubeClient,
+		logger:                 logger,
+		osImage:                osImage,
+		kubeClient:             kubeClient,
+		trunkSupportDiscoverer: installosp.CheckNetworkExtensionAvailability,
```
Comment (pierreprinetti) [inline pkg/controller/machinepool/openstackactuator.go:52]: nit: the function can be scoped to only care about the Trunk extension, if that's all we're interested about:

```Go
type OpenStackActuator struct {
	// ...
	trunkSupportDiscoverer func(cloud string, opts *clientconfig.ClientOpts) (bool, error)
```

```Go
actuator := &OpenStackActuator{
	// ...
	trunkSupportDiscoverer: func(cloud string, opts *clientconfig.Clientopts) (bool, error) {
		return installosp.CheckNetworkExtensionAvailability(cloud, "trunk", opts)
	},
```

Comment (pierreprinetti) [reply pkg/controller/machinepool/openstackactuator.go:52]: I also think you should use what your team is used to! :)

Comment (lleshchi) [reply pkg/controller/machinepool/openstackactuator.go:52]: I do think the current approach is cleaner in terms of readability

**Comment by 2uasimojo** [comment]
/cc @patrickdillon 

would you please have a look at the third commit, which incorporates the new CAPI SG filters from https://github.com/openshift/installer/pull/8006

File: pkg/controller/machinepool/awsactuator_test.go [hunk 2]
```diff
@@ -690,14 +681,40 @@ func TestAWSActuator(t *testing.T) {
...
+						assert.Equal(t, "vpc-id", awsProvider.SecurityGroups[i].Filters[1].Name)
+						if assert.Equal(t, 1, len(awsProvider.SecurityGroups[i].Filters[1].Values), "unexpected number of Values for vpc-id Filter") {
+							assert.Equal(t, customVPC, awsProvider.SecurityGroups[i].Filters[1].Values[0])
+						}
+					}
+				}
+			}
 			if test.expectedSGIDs != nil {
-				if assert.Equal(t, len(test.expectedSGIDs), len(awsProvider.SecurityGroups)-1, "expected n-1 SecurityGroups holding IDs") {
+				if assert.Equal(t, len(test.expectedSGIDs), len(awsProvider.SecurityGroups)-len(defaultSGFilterValues), "expected %d-%d SecurityGroups holding IDs", len(awsProvider.SecurityGroups), len(defaultSGFilterValues)) {
```
[the remark this answers is not in the document]
Comment (2uasimojo) [reply pkg/controller/machinepool/awsactuator_test.go]: Okay, I changed it, not sure if this is what you're looking for...

File: pkg/installmanager/installmanager_test.go [hunk 3]
```diff
@@ -858,9 +920,13 @@ spec:
 				err = json.Unmarshal(machineSet.Spec.Template.Spec.ProviderSpec.Value.Raw, &awsMachineTemplate)
 				assert.NoError(t, err, "expected to be able to decode AWSMachineProviderConfig")
 
-				assert.Contains(t, awsMachineTemplate.SecurityGroups[0].Filters[0].Values, "test-security-group", "expected test-security-group to be configured within Security Group Filters in AWSMachineProviderConfig")
-				assert.Equal(t, awsMachineTemplate.SecurityGroups[0].Filters[1].Name, "vpc-id", "expected an vpc-id named filter to be configured within Security Group Filters in AWSMachineProviderConfig")
-				assert.Contains(t, awsMachineTemplate.SecurityGroups[0].Filters[1].Values, "testvpc123", "expected testvpc123 to be configured within Security Group Filters in AWSMachineProviderConfig")
+				if assert.Equal(t, tc.expectModified, len(awsMachineTemplate.SecurityGroups), "unexpected number of security groups") {
+					for i := 0; i < tc.expectModified; i++ {
+						assert.Contains(t, awsMachineTemplate.SecurityGroups[i].Filters[0].Values, "test-security-group", "expected test-security-group to be configured within Security Group Filters in AWSMachineProviderConfig")
```
Comment (2uasimojo) [inline pkg/installmanager/installmanager_test.go:925]: Need to verify that this contains the original name (`test-czcpt-worker-sg`) as well!

**Pull request merged**